enhancements/aws-permissions-check-bypass.md

```diff
@@ -0,0 +1,269 @@
+installer are queried/checked for appropriate permissions to verify that:
+1. Can be used to complete an installation (the installer set of required permissions)
+1. The in-cluster cloud-credential-operator (CCO) can successfully operate in
+   either 'mint' or 'passthrough' mode.
+If any of these conditions are not met, the install will fail and explain which
+permissions were deemed to be missing.
+
+When an AWS account is configured with Service Control Policies (SCP) the
+permissions checking/simulation APIs  can provide incorrect results depending on
+the contents of the SCPs. Service Control Policies are typically used to deny
+certain API calls unless a condition (or set of conditions) is met.  For
+example, if user `openshift` has a policy attached that allows `ec2:*`, but the
+SCP at the account level denies `ec2:*` unless the API calls are made against
+region `us-east-1`, then the `openshift` user would receive errors when making
+EC2 API calls outside of region `us-east-1`.
+
+Attempting to validate whether user `openshift` has permissions to perform
+`ec2:DescribeInstances` against region `us-east-1` will result in a
+determination that the user `openshift` cannot succesfully perform the API call,
+even though when the actual `ec2:DescribeInstances` call works (as long as it
+is against the allowed region).
+
+In order to acomodate installing OpenShift in these environments, a way is
+needed for the individual performing the installation to indicate that these
+permissions checks should be skipped.
+
+Additionally, a mechanism is needed to indicate to the cloud-credential-operator
+to force it into either `mint` or `passthrough` mode, so that it too can avoid
+attempting to validatate permissions.
+
+### Goals
+
+Enable successfull installation and operation of OpenShift in these AWS accounts
+where the results of permissions simulations cannot be relied upon.
+
+### Non-Goals
+
+Not looking to write complex policy introspection to implement what should
+already be performed by the AWS permissions simulation API.
+
+## Proposal
+
+### Installer
+Introduce an environment variable named
+`OPENSHIFT_INSTALL_ASSUME_PERMISSIONS_MODE` that can be leveraged by users to
+indicate that the installer should not concern itself with determining whether
+the credentials provided are sufficient for installation and for
+cloud-credential-operator.
+
+The installer will then take the value of this variable to build a ConfigMap
+manifest for the CCO so that it too knows what credentials mode it should
+operator under.
+
+### cloud-credential-operator
+Extend the existing ConfigMap that CCO uses, to allow indicating that the
+permissions checking should be bypassed, and to indicate whether `mint` or
+`passthrough` mode should be assumed.
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cloud-credential-operator-config
+  namespace: openshift-cloud-credential-operator
+  annotations:
+    release.openshift.io/create-only: "true"
```

Comment:
hmm.. this makes it seem like CVO already creates this object.

having the installer also push this configmap means there are 2 CREATE only writers for this object, installer during bootstrap can only do create only operations and no updates. This is going to cause race and therefore we need to decide what on one actor.

Reply:
If a user injects a manifest prior to create-cluster does that show up with this annotation?